flink-core/src/main/java/org/apache/flink/configuration/SecurityOptions.java

```diff
@@ -108,6 +108,19 @@
                                     + " (for example, `Client,KafkaClient` to use the credentials for ZooKeeper authentication and for"
                                     + " Kafka authentication)");
 
+    @Documentation.Section(Documentation.Sections.SECURITY_AUTH_KERBEROS)
+    public static final ConfigOption<Boolean> KERBEROS_FETCH_DELEGATION_TOKEN =
+            key("security.kerberos.fetch.delegation-token")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription(
+                            "Indicates whether to fetch delegation token for external services the Flink job needs to contact. "
+                                    + "Only HDFS and HBase are supported, and only works for flink-yarn at the moment. "
+                                    + "If true, Flink will fetch HDFS/HBase delegation tokens and inject into Yarn AM container. "
+                                    + "If false, Flink will assume that the job has delegation tokens and will not fetch them. "
+                                    + "This applies to submission mechanisms like Oozie, which will obtain delegation tokens "
```

Comment:
Could you please elaborate what it means by saying "This applies to submission mechanisms like Oozie"?  IIUC, Flink can't control how Oozie submits jobs, right?

Reply:
IMHO, `This applies to submission mechanisms like Oozie` seems to imply that Oozie also respects this config option. Alternatively, maybe we can say something like: `You may want to disable this option, if you rely on submission mechanisms, e.g. Apache Oozie, to handle delegation tokens`.

Reply:
Good idea.